packages/@aws-cdk/aws-eks/lib/user-data.ts

```diff
@@ -12,8 +12,10 @@ export function renderUserData(clusterName: string, autoScalingGroup: autoscalin
 
   const extraArgs = new Array<string>();
 
-  if (options.useMaxPods) {
-    extraArgs.push('--use-max-pods');
+  if (options.useMaxPods === true) {
+    extraArgs.push(`--use-max-pods`);
+  } else if (options.useMaxPods === false) {
+    extraArgs.push(`--use-max-pods ${options.useMaxPods}`);
```

Comment:
Can we simply always pass this parameter and reduce complexity?

Reply:
Yes, I think we could, but there are some concerns about that.

`useMaxPods` has 3 cases:
 - `true`: same behavior as before
 - `false`: fixed behavior which passing `--use-max-pods false` option
 - `undefined`: default value and I assume that would generate no `--use-max-pods` option

Because all tests assume `undefined` value (which imply `true` of course) and assert no `--use-max-pods` option, I am not intended to make such big change (need discussion) at that time.

If we default to `true` (same as bootstrap.sh), all tests require adding `--use-max-pods` to them and all generated templates would include `--use-max-pods`; if we default to `false`, still we have to add `--use-max-pods false` to all tests and `false` is also a bad default which conflict with bootstrap.sh option.

It's a potential breaking changes, so I would prefer preserving behavior. However, we can discuss and still make such change.

Reply:
I think we should just make `true` the default, always render `--use-max-pods true` and update all test expectations.

Reply:
Ok, I will update them later.